### PR TITLE
convert src/pages/components to TS 

### PR DESCRIPTION
**Pull request by sir-radar**
## Describe your changes
This PR converts all components in the src/pages/components directory to TypeScript by updating the file extensions from .js to .tsx and making necessary changes like adding props interfaces, upgrading packages, updating config files, and creating new files. These changes help improve the maintainability and type safety of the codebase. No logical changes have been made to the files; only the file extensions and necessary type updates have been applied.

Some changes and reasons behind them:

 - Upgraded  chakra-ui/react from  1.7.4 to 2.5.5 to fix `Expression produces a union type that is too complex to represent` error
 
 - Removed `purge` config option from `tailwind.config.js` to fix styling issues I experienced and `content` is recommended 
 
 - Added `svg` declaration in `types.d.ts` to add `svg` file import support
    

## Issue ticket number and link
Closed [#669](https://github.com/nkowaokwu/igbo_api/issues/669)

## Motivation and Context
The motivation behind this change is to enhance the codebase by leveraging TypeScript's static typing. By converting the src/pages/components files to TypeScript, we get to enjoy the benefits of typescript during development and improve the overall robustness of the code.

## How Has This Been Tested?
No tests were written for this PR since it primarily involves file extension changes.

## Screenshots (if appropriate):
No visual changes have been made

### CODE REVIEW

File: src/pages/components/Input/Input.tsx
Comment (ijemmao) [inline src/pages/components/Input/Input.tsx:8]: is it possible to use interface?

File: src/pages/components/MentionedIn/MentionedIn.tsx
Comment (ijemmao) [inline src/pages/components/MentionedIn/MentionedIn.tsx:30]:
```suggestion
const MentionedIn = () => {
```

File: src/pages/components/Navbar/Navbar.tsx
Comment (ijemmao) [inline src/pages/components/Navbar/Navbar.tsx:13]:
```suggestion
const Navbar = ({ to = '/', transparent }: NavBarPropsInterface) => {
```

File: src/pages/components/Navbar/SubMenu.tsx
Comment (ijemmao) [inline src/pages/components/Navbar/SubMenu.tsx:14]:
```suggestion
const SubMenu = ({ isVisible, transparent }: SubMenuPropsInterface) => {
```

File: src/pages/components/Statistics/Stat.tsx
Comment (ijemmao) [inline src/pages/components/Statistics/Stat.tsx:14]:
```suggestion
const Stat = ({ value, header, exact = true, children }: StatPropsInterface) => {
```

File: src/pages/components/Statistics/Statistics.tsx
Comment (ijemmao) [inline src/pages/components/Statistics/Statistics.tsx:25]:
```suggestion
const Statistics = ({
```

**Review by ijemmao — CHANGES_REQUESTED** [review]
great changes! a lot of small comments, but this is really close to getting merged~

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:8]: Hi @ijemmao, with your suggestion here, I got:  `Property 'icon' does not exist on type 'ButtonProps'.` error.

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:12]: `eslint-config-airbnb`  gives `Function component is not a function declaration` error when this change is made. The possible solution so far is to downgrade it from the current version `19.0.4`   to `18.1.0` or leave it as it is. What do you think?

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:8]: `React.ReactNode` could also be used here but I am not sure that best describes an icon.

Comment (ijemmao) [reply src/pages/components/Card/Card.tsx:8]: it would be `ButtonProps['rightIcon']` or `ButtonProps['leftIcon']` - the main focus being that you want to extend from the predefined `ButtonProps` type so you don't have to define these types by hand

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:12]: Hi @ijemmao, I left a comment here.

**Comment by ijemmao** [comment]
@sir-radar are you still working on this PR?

**Comment by sir-radar** [comment]
Hi @ijemmao, yes I am still on this. Could you please take a look at my comment [here](https://github.com/nkowaokwu/igbo_api/pull/699#discussion_r1241745215)

File: src/pages/components/Input/Input.tsx
Comment (sir-radar) [reply src/pages/components/Input/Input.tsx:8]: In this case, it is best to use `type`.

File: src/pages/components/Card/Card.tsx
Comment (ijemmao) [reply src/pages/components/Card/Card.tsx:11]: that's surprising given we use this syntax in other components - https://github.com/nkowaokwu/igbo_api/blob/master/src/pages/components/Footer/Footer.js#L16

can you send a snippet of your change so I can see how you wrote your code

Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:11]: Screenshot 2023-07-06 at 00 41 10

Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:11]: A link to an article on the same issue https://dev.to/andrewallison/react-function-component-is-not-a-function-declaration-58oc

**Comment by ijemmao** [comment]
@sir-radar let me know if you need help with anything!

**Comment by sir-radar** [comment]
Hi @ijemmao, added some new comments [here](https://github.com/nkowaokwu/igbo_api/pull/699#discussion_r1241745215)

**Comment by sir-radar** [comment]
Hello @ijemmao, please give me your opinion about this [here](https://github.com/nkowaokwu/igbo_api/pull/699#discussion_r1241745215) so this PR can be merged

File: src/pages/components/Card/Card.tsx
Comment (ijemmao) [reply src/pages/components/Card/Card.tsx:11]: @sir-radar you'll probably need to remove the `return` statement and do something like

```jsx
const Card = (props) => (
  <>JSX</> // notice how the return keyword isn't getting used
)
```

**Comment by ijemmao** [comment]
@sir-radar just responded!

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:11]: I don't think that is the issue really. As I pointed out in one of my previous comments,  the current `eslint-config-airbnb` version `19.0.4` is the origin of the eslint issue. I will go ahead and downgrade it to a working version `18.1.0` to get this PR merged.

**Comment by sir-radar** [comment]
Hi @ijemmao, I just pushed some changes.

File: src/pages/components/Input/Input.tsx
Comment (ijemmao) [reply src/pages/components/Input/Input.tsx:8]: why? it seems like you don't need to create the `Ref` type and you can directly use the `HTMLInputElement` interface

**Review by ijemmao — COMMENTED** [review]
look really good! just one last comment

**Comment by sir-radar** [comment]
Hi @ijemmao, all outstanding comments are now resolved.

**Comment by sir-radar** [comment]
Hello @ijemmao, I am still waiting on your approval for this PR

**Comment by ijemmao** [comment]
@sir-radar apologies for the late response- i'm reviewing your work now

File: package.json
Comment (ijemmao) [inline package.json:80]: updating package versions should happen in a separate PR, please revert this line change

File: package.json
Comment (ijemmao) [inline package.json:166]: updating package versions should happen in a separate PR, please revert this line change

File: src/pages/components/Card/Card.tsx
Comment (ijemmao) [inline src/pages/components/Card/Card.tsx:30]: ideally we wouldn't want to change anything within these files except if it relates to types

File: src/pages/components/Demo/Demo.tsx
Comment (ijemmao) [inline src/pages/components/Demo/Demo.tsx:20]: if `searchWord` is always expected to be a string, why are we supporting the `string[]` type? let's just have:

```ts
useState(searchWord || '')
```

File: src/pages/components/Demo/Demo.tsx
Comment (ijemmao) [inline src/pages/components/Demo/Demo.tsx:96]: again, i don't think we should be changing the order of className values (unless this is automatically done by prettier)

File: src/pages/components/Footer/Footer.tsx
Comment (ijemmao) [inline src/pages/components/Footer/Footer.tsx:1]: please remove

File: src/pages/components/Input/Input.tsx
Comment (ijemmao) [inline src/pages/components/Input/Input.tsx:12]: good call for adding the comment for now, can you create a follow-up Issue to address removing this comment and stop spreading `rest`?

File: src/pages/components/Navbar/Navbar.tsx
Comment (ijemmao) [inline src/pages/components/Navbar/Navbar.tsx:13]: why are we providing the default value `'/'`? if we're providing a default value, then I think the `to` in `NavBarPropsInterface` should be optional

File: src/pages/components/Navbar/SubMenu.tsx
Comment (ijemmao) [inline src/pages/components/Navbar/SubMenu.tsx:1]: please remove

File: src/pages/components/Statistics/Stat.tsx
Comment (ijemmao) [inline src/pages/components/Statistics/Stat.tsx:2]: please remove

**Review by ijemmao — CHANGES_REQUESTED** [review]
this is absolutely amazing!! we're getting closer - I'm going on vacation from aug 17 to 24 so i won't be able to review during that time but when I'm back i can look over your updates

File: src/pages/components/Demo/Demo.tsx
Comment (sir-radar) [reply src/pages/components/Demo/Demo.tsx:96]: This is automatically done by prettier

File: src/pages/components/Card/Card.tsx
Comment (sir-radar) [reply src/pages/components/Card/Card.tsx:30]: This is automatically done by prettier

**Comment by ijemmao** [comment]
@sir-radar are you still working on this PR?

**Comment by sir-radar** [comment]
Hi @ijemmao, yes I am. Just tied up with work. I will push the changes requested in a few days.

**Comment by ijemmao** [comment]
@sir-radar hope you've been doing well! I wanted to check in to see if there are any updates on this PR?

**Pull request closed without merging**
